idb/postgres/postgres_migrations.go

```diff
@@ -257,50 +278,165 @@ func m3acfgFixAsyncInner(db *PostgresIndexerDb, state *MigrationState, assetIds
 	return -1, nil
 }
 
-func m3acfgFixFlush(db *PostgresIndexerDb, state *MigrationState, tx *sql.Tx, out [][]interface{}) error {
+// m4cumulativeRewardsDBUpdate adds the new rewardstotal column to the account table.
+func m4cumulativeRewardsDBUpdate(db *PostgresIndexerDb, state *MigrationState) error {
+	sqlLines := []string{
+		`ALTER TABLE account ADD COLUMN rewardstotal bigint NOT NULL DEFAULT 0`,
+	}
+	return sqlMigration(db, state, sqlLines)
+}
+
+const cumulativeRewardsUpdateErr = "cumulative rewards migration error"
+
+// m5accountCumulativeRewardsUpdate computes the cumulative rewards for each account one at a time. This is a BLOCKING
+// migration because we don't want to handle the case where accounts are actively transacting while we fixup their
+// table.
+func m5accountCumulativeRewardsUpdate(db *PostgresIndexerDb, state *MigrationState) error {
+	db.log.Println("account cumulative rewards migration starting")
+
+	options := idb.AccountQueryOptions{}
+	if len(state.NextAccount) != 0 {
+		var address sdk_types.Address
+		copy(address[:], state.NextAccount)
+		db.log.Println("after " + address.String())
+		options.GreaterThanAddress = state.NextAccount[:]
+	}
+
+	accountChan := db.GetAccounts(context.Background(), options)
+
+	batchSize := 500
+	// loop through all of the accounts, update them in batches of batchSize.
+	accounts := make([]string, 0, batchSize)
+	for acct := range accountChan {
+		if acct.Error != nil {
+			err := fmt.Errorf("%s: problem querying accounts: %v", cumulativeRewardsUpdateErr, acct.Error)
+			db.log.Errorln(err.Error())
+			return err
+		}
+
+		accounts = append(accounts, acct.Account.Address)
+
+		if len(accounts) == batchSize {
+			db.log.Println("Processing batch of accounts.")
```

Comment:
nit - consider logging the batch number (e.g. 5/2303) in order to get meaningful data and to avoid a long list of "Processing batch of accounts."